src/utils.ts

```diff
@@ -55,16 +55,12 @@ export function URI(url: string | URI = '') {
 export const isDictionary = (maybeDictionary: unknown): maybeDictionary is Dictionary<unknown> =>
   isObjectLike(maybeDictionary);
 
-const getLocalRefValue = (document: unknown, $ref: string): unknown => {
-  return get(document, pointerToPath($ref));
-};
-
 export const maybeResolveLocalRef = (document: unknown, target: unknown): unknown => {
-  if (hasRef(target) && isLocalRef(target.$ref)) {
+  if (isDictionary(document) && hasRef(target) && isLocalRef(target.$ref)) {
     try {
-      return getLocalRefValue(document, target.$ref);
+      return resolveInlineRef(document, target.$ref);
     } catch {
-      return null;
+      return target;
```

Comment:
Here is why I decided to do it
https://github.com/stoplightio/platform-internal/pull/5264/files#r570212260
Feels like having retaining broken $refs may actually provide some value.